### ISSUE TITLE
CNI Plugin not respecting API location from kubeconfig

### ISSUE THREAD

**Issue by caseydavenport** [projectcalico/calico-cni#171]
I'm seeing the following error with Calico CNI version v1.4.0:

```
  2m            1s              128     {kubelet 172.18.18.102}                 Warning         FailedSync      Error syncing pod, skipping: failed to "SetupNetwork" for "kube-dns-v19-a5efy_kube-system" with SetupNetworkError: "Failed to setup network for pod \"kube-dns-v19-a5efy_kube-system(8123784e-7605-11e6-8cf0-080027c23b90)\" using network plugins \"cni\": Get http://localhost:8080/api/v1/namespaces/kube-system/pods/kube-dns-v19-a5efy: dial tcp [::1]:8080: getsockopt: connection refused; Skipping pod"
```

My CNI config looks like so:
```
{
    "name": "k8s-pod-network",
    "type": "calico",
    "etcd_endpoints": "http://127.0.0.1:2379",
    "log_level": "info",
    "ipam": {
        "type": "calico-ipam"
    },
    "policy": {
        "type": "k8s"
    },
    "kubernetes": {
        "kubeconfig": "/etc/kubernetes/worker-kubeconfig.yaml"
    }
}
```

and the referenced kubeconfig looks like this:

```
core@k8s-node-01 ~ $ cat /etc/kubernetes/worker-kubeconfig.yaml
apiVersion: v1
kind: Config
clusters:
- name: local
  cluster:
    server: http://172.18.18.101:8080
users:
- name: kubelet
contexts:
- context:
    cluster: local
    user: kubelet
  name: kubelet-context
current-context: kubelet-context
```

I had seen this working with the `latest` CNI plugin just a day or so before v1.4.0 was released.

**Comment by dkoshkin** [comment]
I'm seeing the same thing running v1.4.0 and k8s v1.3.6. The same config file works on v1.3.1
Setting `k8s_api_root` did not seem to fix it either.

**Comment by anthonyhaussman** [comment]
Seems to be broken since k8s v1.3.4 : 
 [https://github.com/coreos/coreos-kubernetes/issues/600](url)

**Comment by reAsOn2010** [comment]
Same issue here, kubelet keep using default value of k8s apiserver address

**Comment by caseydavenport** [comment]
I've re-tested this from master after #173 was merged and kubeconfig is working as expected.

**Comment by caseydavenport** [comment]
Fixed in release v1.4.1
